src/test/kotlin/org/taktik/icure/services/external/rest/shared/controllers/core/DocumentControllerEndToEndTest.kt

```diff
@@ -0,0 +1,297 @@
+package org.taktik.icure.services.external.rest.shared.controllers.core
+
+import io.kotest.core.spec.style.StringSpec
+import io.kotest.matchers.collections.shouldContainExactlyInAnyOrder
+import io.kotest.matchers.collections.shouldHaveSize
+import io.kotest.matchers.longs.shouldBeGreaterThanOrEqual
+import io.kotest.matchers.nulls.shouldNotBeNull
+import io.kotest.matchers.shouldBe
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.toList
+import org.apache.http.HttpHeaders
+import org.springframework.http.HttpStatus
+import org.springframework.web.reactive.function.client.WebClient
+import org.taktik.icure.asynclogic.objectstorage.testutils.key1
+import org.taktik.icure.asynclogic.objectstorage.testutils.key2
+import org.taktik.icure.asynclogic.objectstorage.testutils.key3
+import org.taktik.icure.asynclogic.objectstorage.testutils.sampleUtis
+import org.taktik.icure.services.external.rest.shared.controllers.core.DocumentControllerEndToEndTestContext.DataFactory.*
+import org.taktik.icure.testutils.authorizationString
+import org.taktik.icure.testutils.shouldContainExactly
+import org.taktik.icure.testutils.shouldRespondErrorStatus
+import org.taktik.icure.utils.toByteArray
+
+val client = WebClient.builder()
```

Comment:
To create the client, use environment variables instead : That way, it will let us execute the test on any environment (local / acc / prd). 
By default, we can use the local env as you did